### PR TITLE
10.0 mig webhook

### PR DESCRIPTION
**Pull request by mmequignon**

### CODE REVIEW

File: webhook/README.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=======
+Webhook
+=======
+
+Module to receive .. _global webhooks: https://en.wikipedia.org/wiki/Webhook events.
```
Comment (sebalix) [inline webhook/README.rst]: This part does not render well on the module page (check the runbot), you could rewrite it this way:
```
Module to receive `webhook <https://en.wikipedia.org/wiki/Webhook/>`_ events.
```

**Review by sebalix — CHANGES_REQUESTED** [review]

File: webhook/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        Webhook odoo controller to receive json request and send to
+        driver method.
+        You will need create your webhook with http://0.0.0.0:0000/webhook
+        NOTE: Important use --db-filter params in odoo start.
+        '''
+        # Deprecated by webhook_name dynamic name
+        # webhook = webhook_registry.search_with_request(
+        #     cr, SUPERUSER_ID, request, context=context)
+        webhook = request.env['webhook'].with_env(
+            request.env(user=SUPERUSER_ID)).search(
```
Comment (Garamotte) [inline webhook/controllers/main.py]: `request.env['webhook'].with_env(request.env(user=SUPERUSER_ID))` must be replaced by `request.env['webhook'].sudo()` (this is clearer, shorter and needs less imports).

File: webhook/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                [('name', '=', webhook_name)], limit=1)
+        # TODO: Add security by secret string  or/and ip consumer
+        if not webhook:
+            remote_addr = ''
+            if hasattr(request, 'httprequest'):
+                if hasattr(request.httprequest, 'remote_addr'):
```
Comment (Garamotte) [inline webhook/controllers/main.py]: @moylop260 Do you remember why these `if` statements ?
The `request.httprequest` should always exist in a controller, IIRC.

File: webhook/demo/webhook_demo.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <data noupdate="1">
```
Comment (Garamotte) [inline webhook/demo/webhook_demo.xml]: `<odoo noupdate="1">` and remove the `data` tag.

File: webhook/models/webhook.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        """
+        Method to search of all webhook
+        and return only one that match with remote address
+        and range of address.
+        :param object request: Request object with data of json
+                               and http request
+        :return: object of webhook found or
+                 if not found then return False
+        """
+        for webhook in self.search([('active', '=', True)]):
```
Comment (Garamotte) [inline webhook/models/webhook.py]: `[('active', '=', True)]` is the default domain when an `active` field exists, unless you add `active_test=False` in the context. Any reason to enforce this domain ?

File: webhook/README.rst [hunk 6]
```diff
@@ -0,0 +1,143 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=======
+Webhook
+=======
+
+Module to receive `webhook <https://en.wikipedia.org/wiki/Webhook/>`_ events.
```
Comment (sebalix) [inline webhook/README.rst]: Sorry, the URL is `https://en.wikipedia.org/wiki/Webhook` (without the trailing slash).

**Review by sebalix — CHANGES_REQUESTED** [review]

File: webhook/controllers/main.py (hunk 3 above)
Comment (moylop260) [reply webhook/controllers/main.py]: I think we can replace it with:
```python
if not webhook:
    remote_addr = getattr(request.httprequest, 'remote_addr', None)
    raise exceptions.ValidationError(_(...
```

**Review by moylop260 — APPROVED** [review]
Thanks for migration and improvements.

**Review by sebalix — APPROVED** [review]

**Review by Garamotte — APPROVED** [review]

**Comment by yajo** [comment]
Squash migration commits to merge please.

**Comment by mmequignon** [comment]
@Yajo Done.

**Pull request merged**